docs/en/stack/ml/anomaly-detection/ml-troubleshooting.asciidoc

```diff
@@ -1,118 +1,154 @@
-
-This problem occurs when you have a {dfeed} that contains search or query
-domain specific language (DSL) that was discontinued. For example, if you
-created a {dfeed} query in 5.x using search syntax that was deprecated in 5.x
-and removed in 6.0, you must fix the {dfeed} before you upgrade to 6.0.
-
-*Symptoms:*
-
-* If {ref}/logging.html#deprecation-logging[deprecation logging] is enabled
-before the upgrade, deprecation messages are generated when the {dfeeds} attempt
-to retrieve data.
-* After the upgrade, nodes fail to start and the error indicates that they
-failed to read the local state.
-
-*Resolution:*
-
-Before you upgrade, identify the problematic search or query DSL. In 5.6.5 and
-later, the Upgrade Assistant detects these scenarios. If you cannot fix the DSL
-before the upgrade, you must delete the {dfeed} then re-create it with valid DSL
-after the upgrade.
-
-If you do not fix or delete the {dfeed} before the upgrade, in order to successfully
-start the failing nodes you must downgrade the nodes then fix the problem per
-above.
-
-See also {stack-ref}/upgrading-elastic-stack.html[Upgrading the Elastic Stack].
\ No newline at end of file
+To avoid this problem, manually update the mappings on the {ml} annotations and
+config indices in your old {es} version before you upgrade to 7.9.0. For example:
+
+[source,console]
+--------------------------------------------------
+PUT .ml-annotations-6/_mapping
+{
+  "properties": {
+    "event" : {
+      "type" : "keyword"
+    },
+    "detector_index" : {
+      "type" : "integer"
+    },
+    "partition_field_name" : {
+      "type" : "keyword"
+    },
+    "partition_field_value" : {
+      "type" : "keyword"
+    },
+    "over_field_name" : {
+      "type" : "keyword"
+    },
+    "over_field_value" : {
+      "type" : "keyword"
+    },
+    "by_field_name" : {
+      "type" : "keyword"
+    },
+    "by_field_value" : {
+      "type" : "keyword"
+    }
+  }
+}
+--------------------------------------------------
+// TEST[skip:TBD]
+
+//TBD: Are the same mappings required for the .ml-config index?
+
+If the problem exists in your {ml} annotations index after you upgrade, you must
+reindex it. For example:
+
+[source,console]
+--------------------------------------------------
+# 1. Enable upgrade mode
+POST _ml/set_upgrade_mode?enabled=true&timeout=10m
+
+# 2. . Create a temporary index 
+PUT temp_ml_annotations
+
+# 3. Reindex the `.ml-annotations-6` index into the temporary index
+POST _reindex
+{
+  "source": { "index": ".ml-annotations-6" }, 
+  "dest": { "index": "temp_ml_annotations" }
+}
+
+# 4. Delete the .ml-annotations-6 index
+DELETE .ml-annotations-6
+
+# 5. Wait for .ml-annotations-6 to be recreated
+
+# 6. Reindex the temporary index into the .ml-annotations-6 index
+POST _reindex
+{
+  "source": { "index": "temp_ml_annotations" }, 
+  "dest": { "index": ".ml-annotations-6" }
+}
+
+# 7. Disable upgrade mode
+POST _ml/set_upgrade_mode?enabled=false&timeout=10m
```

Comment:
It would be best to move this between current steps 4 and 5, i.e. 7 -> 5, 5 -> 6, 6 -> 7.

Reply:
Thanks, I've made the same change in the second set of resolution steps too.